### ISSUE TITLE
page buttons in landscape mode for reMarkable 1 are reversed

### ISSUE THREAD

**Issue by hohusum** [koreader/koreader#6842]
* KOReader version: 2020.10.1
* Device: reMarkable1

#### Issue
When viewing a pdf and rotating -90 or +90 degrees logically, and rotating the device physically to match, to view in landscape mode, the paging buttons behave in an illogical manner. The lower button when pressed brings you backward in document (from page 3 to page 2), while the upper button brings you forward in the document (from page 3 to page 4).

This does not seem logical, and is not how readers like Kobo Forma or Kobo Libra work in landscape mode.

#### Steps to reproduce

1. Open pdf document
2. Use footer menu to set rotation to +90 or -90 degrees
3. Rotate the physical device to match the logical rotation set in step 2
4. Press the upper button repatedly to see how you move through the document
5. Press the lower button repeatedly to see how you move through the document

##### `crash.log` (if applicable)
`crash.log` is a file that is automatically created when KOReader crashes. It can
normally be found in the KOReader directory:

* `/mnt/private/koreader` for Cervantes
* `koreader/` directory for Kindle
* `.adds/koreader/` directory for Kobo
* `applications/koreader/` directory for Pocketbook

Android won't have a crash.log file because Google restricts what apps can log, so you'll need to obtain logs using `adb logcat KOReader:I ActivityManager:* AndroidRuntime:* *:F`.

Please try to include the relevant sections in your issue description.
You can upload the whole `crash.log` file on GitHub by dragging and
dropping it onto this textbox.

**Comment by NiLuJe** [comment]
Given where the buttons are located (*both* on the bottom, one on the left, the other on the right), the naive mapping of honoring that placement makes sense.

c.f., https://github.com/koreader/koreader/blob/master/frontend/device/remarkable/event_map.lua

Do see if switching to a full LPg* or RPg* mapping helps, despite making no sense on paper ^^.

**Comment by pazos** [comment]
@NiLuJe: it happens also on android with (bluetooth) keyboard arrows UP/DOWN/LEFT/RIGHT and the emulator between UP/DOWN.

**Comment by NiLuJe** [comment]
@pazos: That's https://github.com/koreader/koreader/blob/74c1813a8211b721f3e19d9a03977f71c2ee515d/frontend/device/input.lua#L119-L125

I vaguely recall that it made sense the last time I checked, but that was a very good long while ago ^^.

**Comment by hohusum** [comment]
What actually happens is that the top button does page forward, the bottom button does page back, whether I am at -90 or +90 degrees of rotation. So the actual behavior seems to be in conflict with the "--NOTE" text in the code you pasted above.

**Comment by NiLuJe** [comment]
That comment refers to a classic Kindle like design with two buttons on the *side*, and not on the bottom like the rM.

**Comment by hohusum** [comment]
My use case is the following: I am right-handed, and read in various European languages. I keep the device in portrait mode as long as the cropping gives a readable text size. If that doesn't work I rotate the device logically and physically. If I am mainly reading, I rotate it such that the buttons are on the right. If I am reading something that requires a lot of notetaking, or working through proofs or calculations on the side, I rotate it such that the buttons are on the left. The most common operation is to page forward. If the lower button in the current orientation is bound to page forward, I can press it with my arm resting on the desk, even when the device is held at an angle or placed on a reading stand. If the upper button in the current orientation is bound to page forward I must raise my entire arm to perform the operation. Ergonomically the lower button placement is better.

**Comment by NiLuJe** [comment]
Agreed, c.f. the earlier edit of my original answer ;). 

> 
> The rM probably warrants a custom `rotation_map` where LPgBack = LPgBack && RPgFwd = LPgFwd @ +90° and.... whatever you think makes sense at -90° (possibly keeping forward on the bottom, so LPgBack = RPgFwd && RPgFwd = RPgBack).

**Comment by zelomel** [comment]
You can invert the buttons: 3rd Menu, Navigation, Invert page turn buttons

**Comment by NiLuJe** [comment]
Or, still waiting for someone with an rM to test & PR that custom `rotation_map` ^^.

**Comment by hohusum** [comment]
Should this be working in the 2020.12 release? I have that and can test.

**Comment by hohusum** [comment]
> Should this be working in the 2020.12 release? I have that and can test.

No, it does not work in 2020.12; the manual invert buttons menu choice works, but NiLuJes solution should do the right thing based on whether clockwise or counter-clockwise 90 degree rotation is chosen.

**Comment by addcninblue** [comment]
Hi everyone -- I had this exact problem, and this fixed it:

In `koreader/frontend/device/input.lua`, I switched `ORIENTATION_LANDSCAPE_ROTATED` and `ORIENTATION_LANDSCAPE` in the `rotation_map`, as follows:

```
rotation_map = {
    [framebuffer.ORIENTATION_PORTRAIT] = {},
    [framebuffer.ORIENTATION_LANDSCAPE_ROTATED] = { Up = "Right", Right = "Down", Down = "Left", Left = "Up", LPgBack = "LPgFwd", LPgFwd = "LPgBack", RPgBack = "RPgFw
    [framebuffer.ORIENTATION_PORTRAIT_ROTATED] = { Up = "Down", Right = "Left", Down = "Up", Left = "Right", LPgFwd = "LPgBack", LPgBack = "LPgFwd", RPgFwd = "RPgBack
    [framebuffer.ORIENTATION_LANDSCAPE] = { Up = "Left", Right = "Up", Down = "Right", Left = "Down" }
},
```

As this is a pretty hacky solution, I don't think this will work for all devices, but it's a decent workaround for our Remarkables.

Hope this helps!
